src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -25,7 +25,7 @@ MPL_STATIC_INLINE_PREFIX uint16_t MPIDI_OFI_am_fetch_incr_send_seqno(MPIR_Comm *
     else
         old_seq = (uint16_t) (uintptr_t) ret;
 
-    seq = old_seq + 1;
+    seq = (uint16_t) (old_seq + 1);
```

Comment:
I'm surprised by this one. Which complier warns on it? Neither clang or gcc-10 on my laptop seem to flag it.

Reply:
You have to set `-Wconversion` to see it. It is currently a mess for mpich internal code, but it will be very nice once we make it clean.

Reply:
I tried a test program with `-Wconversion` and couldn't get it to warn.
```
#include <stdint.h>

int main(void) {
    uint16_t a = 0, b;

    b = a + 1;

    return 0;
}
```

Reply:
I think the compiler can tell it is not out-of-range in this case.